File: ase/misc/word.js

var cb = {
  'Quit' : function() { print("QUIT"); }
};
var word = COM.CreateObject("Word.Application");
COM.Connect(word, cb);
word.Visible = true;
var i = 0;
while (true) {
  COM.DispatchMessage(1000);
  print("sleep " + i);
  i += 1
}
print("DONE");

File: ase/misc/word.lua

local quit_flag = false;
local cb = {
  Quit = function() print("QUIT"); quit_flag = true; end,
};
local word = COM.CreateObject("Word.Application");
COM.Connect(word, cb);
word.Visible = true;
local i = 0;
while (not quit_flag) do
  COM.DispatchMessage(1000);
  print("sleep " .. i);
  i = i + 1;
end
print("DONE");

